lib/req/request.ex

```diff
@@ -181,6 +181,7 @@ defmodule Req.Request do
 
   Any step can call `halt/1` to halt the pipeline. This will prevent any further steps
   from being invoked.
+  You need to return a tuple with an exception or a response. Otherwise the halt is ignored.
```

Comment:
yeah, ignoring halt was definitely a bug. I decided to deprecate halt/1 in favour of halt/2 where we're forced to return response|exception, I just pushed a commit to main.

Even though I didn't merge your PR I really appreciate your effort in general and this little note in particular as it made it obvious the previous API didn't really make sense. Thank you!